### PR TITLE
mail/postfix: add smtp security enforcement and certificate selection

### PR DESCRIPTION
**Pull request by mimugmail**
- Let the user choose whether the SMTP MUST be encrypted or MAY be encrypted
- Let the user choose certficate and CA, looks good under https://de.ssl-tools.net/mailservers

### CODE REVIEW

File: mail/postfix/src/opnsense/mvc/app/controllers/OPNsense/Postfix/forms/general.xml [hunk 1]
```diff
@@ -61,6 +61,24 @@
...
+        <id>general.ca</id>
+        <label>Root CA</label>
+        <type>dropdown</type>
+        <help>Choose the Certificate Authority which signed your certificate.</help>
+    </field>
+    <field>
+        <id>general.smtpclient_security</id>
+        <label>SMTP Client Security</label>
+        <type>dropdown</type>
+        <help>Choose none to disable TLS for sending mail. Set encrypt to enforce TLS security, please don't use this for Internet wide communication as no every server supports TLS yet. Default is may which will use TLS when offered.</help>
```
Comment (fichtner) [inline mail/postfix/src/opnsense/mvc/app/controllers/OPNsense/Postfix/forms/general.xml]: "none"

**Comment by mimugmail** [comment]
As always need chmod 755 for the new php script :)

File: mail/postfix/src/opnsense/mvc/app/controllers/OPNsense/Postfix/forms/general.xml (hunk 1 above)
Comment (fichtner) [reply mail/postfix/src/opnsense/mvc/app/controllers/OPNsense/Postfix/forms/general.xml]: not every server
"may"

Comment (mimugmail) [reply mail/postfix/src/opnsense/mvc/app/controllers/OPNsense/Postfix/forms/general.xml]: man .. you're way too fast :)

Comment (fichtner) [reply mail/postfix/src/opnsense/mvc/app/controllers/OPNsense/Postfix/forms/general.xml]: good to be back!

Comment (mimugmail) [reply mail/postfix/src/opnsense/mvc/app/controllers/OPNsense/Postfix/forms/general.xml]: fixed

**Comment by fichtner** [comment]
Ready for merge?

**Comment by mimugmail** [comment]
Yep :)

**Comment by fichtner** [comment]
thanks!

**Pull request merged**